runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutor.java

```diff
@@ -351,7 +398,115 @@ void updateProgress() {
       }
     }
 
-    private void updateMetrics(BeamFnApi.Metrics metrics) {
+    // Keeping this as static class for this iteration. Will extract to separate file and generalize
+    // when more counter types are added.
+    // todomigryz: define counter transformer factory
+    // that can provide respective counter transformer for different type of counters.
+    // (ie RowCountCounterTranformer, MSecCounterTransformer, UserCounterTransformer, etc)
+    private static class MonitoringInfoToCounterUpdateTransformer {
+
+      private final Map<String, DataflowStepContext> transformIdMapping;
+
+      public MonitoringInfoToCounterUpdateTransformer(
+          final Map<String, DataflowStepContext> transformIdMapping) {
+        this.transformIdMapping = transformIdMapping;
+      }
+
+      // todo: search code for "beam:metrics"... and replace them with relevant enums from
```

Comment:
Should this be fixed before merging?

Reply:
This depends on https://github.com/apache/beam/pull/6799
Will be implemented in further PRs.